### PR TITLE
[SFN] Add Batch Optimised Service Integration

### PR DESCRIPTION
**Pull request by MEPalma**
Depends on https://github.com/localstack/localstack/pull/11197 which implements the new sync integration pattern

## Motivation
Currently the SFN v2 interpreter offers no support for [Batch Optimised Integrations](https://docs.aws.amazon.com/step-functions/latest/dg/connect-batch.html). This PR adds such optimised service integration.

## Changes
- Added optimised service task for batch
- Resolved minor issue in parameters normalisation when these are in a typed list

## Testing
- Positive and negative Snapshot tests are added in Tandem PR

### CODE REVIEW

File: localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_batch.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    ) -> None:
+        batch_client = boto_client_for(
+            region=resource_runtime_part.region,
+            account=resource_runtime_part.account,
+            service="batch",
+        )
+        submission_output: dict = env.stack.pop()
+        job_id = submission_output["JobId"]
+
+        def _has_terminated() -> Optional[dict]:
```
Comment (joe4dev) [inline localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_batch.py:148]: typing nit: we could (at some point) consider migrating to `dict | None` at some point or for new code (now natively supported)

File: localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_batch.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        env.stack.append(termination_output)
+
+    def _sync(
+        self,
+        env: Environment,
+        resource_runtime_part: ResourceRuntimePart,
+        normalised_parameters: dict,
+    ) -> None:
+        match self._get_boto_service_action():
```
Comment (joe4dev) [inline localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_batch.py:192]: Would a simple if, else be easier to read here?

File: localstack-core/localstack/testing/pytest/stepfunctions/fixtures.py [hunk 3]
```diff
@@ -22,6 +22,52 @@ def sfn_snapshot(snapshot):
     return snapshot
 
 
+@pytest.fixture
+def sfn_batch_snapshot(sfn_snapshot):
```
Comment (joe4dev) [inline localstack-core/localstack/testing/pytest/stepfunctions/fixtures.py:26]: Are we collecting all sfn fixtures here for consistency or is there another reason why we should ship this with the community image, assuming it's only used in ext tests?

**Review by joe4dev — APPROVED** [review]
LGTM 👍

**Comment by joe4dev** [comment]
This failing test might be related to the changes 🤔 

```
FAILED tests/aws/services/stepfunctions/v2/services/test_events_task_service.py::TestTaskServiceEvents::test_put_events_base
```

The other BookStore acceptance tests seem to be known flakes.

File: localstack-core/localstack/testing/pytest/stepfunctions/fixtures.py (hunk 3 above)
Comment (MEPalma) [reply localstack-core/localstack/testing/pytest/stepfunctions/fixtures.py:27]: We collect all sfn fixtures here only for consistency

**Pull request merged**